### ISSUE TITLE
ESLint in OpenEMR

### ISSUE THREAD

**Issue by stu01509** [openemr/openemr#3057]
### Describe the problem

Hello All,

I have a suggestion, I consider OpenEMR is a huge project, We have HTML CSS javascript PHP a lot of languages in our project, Why not to use ESlint or define coding style to maintain code quality?

**Comment by thesmallstar** [comment]
Yes, also there are lot files without indentation, useless comments we can also remove that.

I guess sir @bradymiller  had not done it yet, so as to get maximum contributions and not to restrict developers 😅

**Comment by bradymiller** [comment]
hi @stu01509 and @thesmallstar ,
 
We do have a travis check to attempt to keep things somewhat reasonable :)
https://github.com/openemr/openemr/blob/master/.travis.yml#L11
The rules it uses are here:
https://github.com/openemr/openemr/blob/master/ci/phpcs.xml

Prior to instituting this (along with 2 megacommits to convert the codebase at that time), the code styling was literally all over the place.

Always up for improving the standard code styling. Better to institute it from travis (since we are then forced to comply) if possible. The problem I've noted with the current mechanism is it only seem to look at the php code and not the other elements (js, css, html). Are there are tools we can take advantage for those elements?

thanks! -brady

**Comment by bradymiller** [comment]
duh, just realized you mentioned a good tool in the title of the issue:
https://eslint.org/

Definitely feel free to incorporate this. Guessing will involve a megacommit to convert the codebase, then support in travis, then with instructions on how to support in PHPStorm (many developers use this; we actually get free licenses and I have been working on getting more since we have maxed out on those licenses).

**Comment by stu01509** [comment]
@thesmallstar 

Yeah, I think  the ESLint is easily to import to our project,
We can follow the ESLint rules or custom rules, that javascript has the same coding style,
It well really helps to develop and maintain our code quality.

**Comment by stu01509** [comment]
Hi @bradymiller sir

Please assign this issue to me, Thanks.

**Comment by stephenwaite** [comment]
hi @stu01509 , be happy to, thank you

**Comment by tywrenn** [comment]
@stu01509 I actually looked at this as well when I was looking at using stylelint 😂 it's actually really cool. Main thing to keep in mind is to ignore crucial JS system files (specifically gulpfile.js).

**Comment by stu01509** [comment]
Hello @tywrenn 

Thanks for your reminder, I will add the gulpfile.js to the ignore rules :)

**Comment by kojiromike** [comment]
Is this issue not resolved? I see Eslint in the project…

**Comment by bradymiller** [comment]
correct, closing this
